src/VisualStudio/Core/Impl/RoslynVisualStudioWorkspace.cs

```diff
@@ -162,7 +174,10 @@ internal override object GetBrowseObject(SymbolListItem symbolListItem)
             }
 
             var tree = sourceLocation.SourceTree;
+            Contract.ThrowIfNull(tree, "We have a location that was in source, but doesn't have a SourceTree.");
```

Comment:
>"We have a location that was in source, but doesn't have a SourceTree." [](start = 39, length = 71)

I think we should avoid these strings and just use comments. The strings have some runtime overhead and also make the assembly bigger.

Reply:
I wonder if we can use another pattern to get both the benefit of terse syntax and useful error message.

Perhaps if we defined an `enum ContractFailure { LocationInSourceWithoutSourceTree = 1, ... }` and then have overload `Contract.ThrowIfNull(tree, ContractFailure.LocationInSourceWithoutSourceTree);`

The implementation could use the name of the enum field as a message.

Reply:
Closing in favor of #42114.